### ISSUE TITLE
[Op4CTF] Long Jump Rune

### ISSUE THREAD

**Issue by APGRoboCop** [ValveSoftware/halflife#1857]
Hello I've noticed on Op4CTF Linux server that the LJ Rune doesn't appear to work when I duck jump forward. I not sure if all the DeathMatch maps of Op4 with the Long Jump module is affected as well but it should be running fine on op4_crossfire, from what I've recently tested. Also when I switch from a op4ctf_ to a op4_ map the "Time Remaining" HUD appears to remain in DeathMatch maps. And no biggie but is it possible to increase the column width and character limit for the Scoreboard as well please?

> 20:18:15 Protocol version 48
>          Exe version 1.1.2.2/Stdio (valve)
>          Exe build: 17:35:12 Jul  5 2017 (7559)

**Comment by tschumann** [comment]
Yeah I've seen the Time Remaining remain too (even in single player I think).

**Comment by SamVanheer** [comment]
The CTF long jump item sets the `jpj` physics keyvalue to `1`, the original longjump item sets `slj` to `1`. No checks exist for `jpj`, only `slj` as in the SDK.

As for the timer, `CHudFlagIcons::m_bIsTimer` is set by a message function which is only used by CTF gamerules. It is reset in `CHudFlagIcons::Init`, which is called by `CHud::Init`, which is called by `HUD_Init`, which according to the comments is supposed to be called every time the client connects to a server.
This is not true in the current engine build so that change probably broke it.

To fix this, `VidInit` may used, as it is called every time the client connects to a server. `HUD_VidInit` -> `CHud::VidInit` -> `CHudFlagIcons::VidInit`.

**Comment by lith01** [comment]
@SamVanheer I saw all duplicate issues to this one and I tried using a AMX Mod based fix but it won't solve the issue. Is there any outlook to when this fix will be released? All tickets have been marked with fix proposed and even there was set a "milestone" but I do not see any progress after three years.
I do not have Windows servers and would really like to set up some proper OP4 CTF server.

**Comment by SamVanheer** [comment]
As far as i can tell there is nobody actively working on GoldSource engine games so i wouldn't expect any fixes from Valve.

[My Opposing Force SDK](https://github.com/SamVanheer/halflife-updated/wiki) is almost ready for a full release so that could be an alternative option, but it doesn't support crossplay with the original Opposing Force.

**Comment by djearthquake** [comment]
@lith01 The jump aspect runs off fakemeta. Assure the module is working in Amxx please. https://github.com/djearthquake/amxx/blob/main/scripting/gearbox/op4ctf_jumpfix.sma
```c
fm_set_user_longjump(target, true, true)
```
https://forums.alliedmods.net/showthread.php?t=317504 Welcome to the community.
